include/mbedtls/config_psa.h

```diff
@@ -865,6 +874,11 @@ extern "C" {
 #define PSA_WANT_ALG_SHA_512 1
 #endif
 
+#if defined(MBEDTLS_SHA3_C)
+#define MBEDTLS_PSA_BUILTIN_ALG_SHA_3 1
+#define PSA_WANT_ALG_SHA_3 1
```

Comment:
Maybe I'm missing something, but are we not defining `PSA_WANT_ALG_SHA_3` in line 117 already in the same file?

Reply:
That's happening for different reasons. Line 117 is:
```
/* Hashes that are built in are also enabled in PSA.
 * This simplifies dependency declarations especially
 * for modules that obey MBEDTLS_USE_PSA_CRYPTO. */
```
This is done unconditionally, only for hash functions.

Line 877 is:
```
/* Infer PSA requirements from Mbed TLS capabilities */
```
gated by `!defined(MBEDTLS_PSA_CRYPTO_CONFIG)`: builds using the legacy API to configure cryptographic mechanisms. For all legacy cryptographic mechanisms, enable the corresponding PSA API.

https://github.com/Mbed-TLS/mbedtls/pull/7615 will make this clearer.

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/7615 makes it better. :)